ksmt-cvc5/src/main/kotlin/org.ksmt.solver.cvc5/KReCurrentScopeInternalizer.kt

```diff
@@ -0,0 +1,78 @@
+package org.ksmt.solver.cvc5
+
+import org.ksmt.KContext
+import org.ksmt.decl.KDecl
+import org.ksmt.expr.KApp
+import org.ksmt.expr.KArrayLambda
+import org.ksmt.expr.KConst
+import org.ksmt.expr.KExistentialQuantifier
+import org.ksmt.expr.KExpr
+import org.ksmt.expr.KFunctionApp
+import org.ksmt.expr.KFunctionAsArray
+import org.ksmt.expr.KUniversalQuantifier
+import org.ksmt.expr.rewrite.KExprUninterpretedDeclCollector
+import org.ksmt.expr.transformer.KNonRecursiveTransformer
+import org.ksmt.sort.KArraySort
+import org.ksmt.sort.KBoolSort
+import org.ksmt.sort.KSort
+
+
+private class KReCurrentScopeInternalizer(
+    private val cvc5Ctx: KCvc5Context, ctx: KContext
+) : KNonRecursiveTransformer(ctx) {
+    override fun <T : KSort, A : KSort> transformApp(expr: KApp<T, A>): KExpr<T> {
+        if (cvc5Ctx.findCurrentScopeInternalizedExpr(expr) != null)
```

Comment:
You can't stop non-recursive transformation this way.

Reply:
After (Expression transformation skip)[https://github.com/UnitTestBot/ksmt/pull/78] push we use ```override fun <T : KSort> exprTransformationRequired(expr: KExpr<T>): Boolean = expr !in currentScopeExpressions```